sql/memo/join_order_builder.go

```diff
@@ -589,6 +589,12 @@ func (j *joinOrderBuilder) constructJoin(
 		rel = &SemiJoin{b}
 	case plan.JoinTypeAnti:
 		rel = &AntiJoin{b}
+	case plan.JoinTypeLateralCross:
```

Comment:
These are all treated the same in reordering, can they be one `ApplyJoin` with a `*plan.JoinNode`? Reordering does need the nodes schema maybe, building the sql.Node again still needs to fix indexes depending on degenerate (cross) or not.